packages/core/src/Synchronizer.ts

```diff
@@ -681,25 +786,29 @@ export class Synchronizer extends EventEmitter {
      * @param epoch The epoch query
      * @returns The number of the global state tree leaves
      */
-    async numStateTreeLeaves(epoch: number) {
+    async numStateTreeLeaves(
+        epoch: number,
+        attesterId: bigint | string = this.attesterId
+    ) {
         return this._db.count('StateTreeLeaf', {
             epoch: epoch,
-            attesterId: this.attesterId.toString(),
+            attesterId: toDecString(attesterId),
         })
     }
 
     // unirep event handlers
 
-    async handleStateTreeLeaf({ event, db, decodedData }: EventHandlerArgs) {
+    async handleStateTreeLeaf({
+        event,
+        db,
+        decodedData,
+        _attesterId,
```

Comment:
I don't like taking `attesterId` as an argument here. Instead we should use the class variables.

Reply:
[f998c42](https://github.com/Unirep/Unirep/pull/367/commits/f998c420c425ea4ac08f5a191967cd231c9e64b6)